File: python/drivers/EXPORTER.py
class EXPORTER(object):
    def __init__(self, filename):
        self.fileRef = open("%s.csv" % filename, 'w')
    # end __init__

    def writeData(self, dataList):
        for i in range(len(dataList)):
            if isinstance(dataList[i], complex):
                dataStr = str(dataList[i].real)
            else:
                dataStr = str(dataList[i])
            
            if i == 0:
                self.fileRef.write(dataStr)
            else:
                self.fileRef.write(",%s" % dataStr)
        self.fileRef.write("\n")
    # end writeData

    def close(self):
        self.fileRef.close()
    # end close

File: matlab/drivers/EXPORTER.m
classdef EXPORTER
    properties
        fId
    end
    
    methods
        function obj = EXPORTER(filename)
            obj.fId = fopen([filename '.csv'], 'wt');
        end
        
        function writeData(obj, dataList)
            for i = 1:length(dataList)
                if ischar(i)
                    data = dataList{i};
                else
                    data = num2str(dataList{i});
                end
                
                if i == 1
                    fprintf(obj.fId, '%s', data);
                else
                    fprintf(obj.fId, ',%s', data);
                end
            end
            
            fprintf(obj.fId, '\n');
        end
        
        function close(obj)
            fclose(obj.fId);
        end
    end
end